File: Dynamic Programming/Fibonacci Series/FibonacciWithDPCode/BottomUp/FibonacciWithDP_BU.java
class Fib
{
    private int[] _dp;
    
    public Fib(int n)
    {
        _dp = new int[n+1];
    }
    
    public int getNthFibonacciNumber(int n)
    {
        _dp[0] = 0;
        _dp[1] = 1;

        for(int i = 2; i < n+1; i++)
        {
            _dp[i] = _dp[i-1] + _dp[i-2];
        }

        return _dp[n];
    }
}

public class Main
{
	public static void main(String[] args) 
    {
		int n = 8;
		Fib fib = new Fib(n);
        System.out.println(fib.getNthFibonacciNumber(n));
	}
}

File: Dynamic Programming/Fibonacci Series/FibonacciWithDPCode/BottomUp/FibonacciWithDP_BU.py
class Fib:
    def __init__(self, n) -> None:
        self._dp = [-1] * (n+1)

    def getNthFibonacciNumber(self, n):
        self._dp[0] = 0
        self._dp[1] = 1
        
        for i in range(2, n+1):
            self._dp[i] = self._dp[i-1] + self._dp[i-2]

        return self._dp[n]
    

n = 8
fib = Fib(n)
print(fib.getNthFibonacciNumber(n))